### PR TITLE
Implement JWT Authentication for regular users

### LINKED ISSUE #8: user login
user can login to system

### PR DESCRIPTION
**Pull request by Dragonfly911117**
This PR primarily introduces JSON Web Token (JWT) authentication to protect the content within the FastAPI application. Here are the detailed changes:

1. JWT Authentication: Authentication mechanisms have been implemented to manage gathering/updating of accounts, shops, products, and image uploads. User permissions are enforced, ensuring authorized access and enhancing system security.

2. Misc: Improved error handling and file updates, including clearer error messages aligned with HTTP status code conventions. The README file now provides improved instructions for setting environment variables, aiding new contributors in project setup.

The primary focus of this PR is the implementation of JWT authentication for account/shop/product/image operations while improving code efficiency. These changes bolster the security, functionality, and maintainability of the FastAPI application.

## 中文好讀版:

這個 PR 主要是引入 JSON Web Token (JWT) 認證，以保護 FastAPI 應用程式中該被保護的東西。以下是變更的詳細內容：

1. JWT 認證: 已實施認證機制來管理帳戶、商店與商品的收集/更新以及圖片上傳。強制執行使用者許可權，確保授權訪問，提升系統安全性。

2. 其他增強功能: 改進了錯誤處理和檔案更新，包括與 HTTP 狀態碼慣例相符的更清晰的錯誤訊息。 README 檔案現在提供了改進的環境變數設定說明，幫助新貢獻者進行專案設定。

這個 PR 的主要焦點是實現帳號、商店、商品與圖片的 JWT 認證，同時改進了程式效率。這些變更增強了 FastAPI 應用程式的安全性、功能性和可維護性。

### CODE REVIEW

**Comment by Dragonfly911117** [comment]
While this PR shares some similarities with #34, it also includes JWT implementations which will greatly help in addressing issues like those seen in #37.

File: model/account.py [hunk 1]
```diff
@@ -35,7 +35,6 @@ class ReturnCreateAccount(SuccessModel):
 
 @as_form
 class UpdateAccountForm(BaseModel):
-    account_uuid: str
```
Comment (Vghxv) [inline model/account.py:38]: Q1

File: utils/db_process.py [hunk 2]
```diff
@@ -51,8 +55,11 @@ def execute_sql(sql, param: Optional[Tuple] = None, fetch: bool = False) -> Unio
 
     return result
 
-def get_all_results(sql, param: Optional[Tuple] = None) -> Union[Dict, bool]:
-    return execute_sql(sql, param, fetch=True)
+def get_all_results(sql: str, param: Optional[Tuple] = None) -> Union[Dict, bool]:
+    fetch = True
+    if sql.split()[0].upper() == "UPDATE" or sql.split()[0].upper() == "DELETE":
+        fetch = False
+    return execute_sql(sql, param, fetch=fetch)
```
Comment (Vghxv) [inline utils/db_process.py]: programmer should know what to use for what sql operation, i think select sth and create/update operation do not have that much similarity.
btw, in function `execute_sql` actually do not need to connection.commit() when fetch == true(i.e. select) bcz database do not need update

File: utils/db_process.py [hunk 3]
```diff
@@ -86,3 +93,11 @@ async def if_exists_in_db(table_name, column_name, value) -> bool:
...
     assert result[0]["UUID_Exists"] in [0, 1]
     return result[0]["UUID_Exists"] == 1
+
+async def if_one_owns_the_other(
+        table_name: str, col1: str, val1: str, col2: str, val2: str
+) -> bool:
+    sql = f"SELECT EXISTS (SELECT 1 FROM {table_name} WHERE  {col1} = %s AND {col2} = %s) AS 'Exists';"
+    result = get_all_results(sql, (val1, val2))
+    assert result[0]["Exists"] in [0, 1]
+    return result[0]["Exists"] == 1
```
Comment (Vghxv) [inline utils/db_process.py:100]: although i hate wrapper, i think this is acceptable bcz check database do confuse ppl a bit

File: router/shop.py [hunk 4]
```diff
@@ -1,128 +1,106 @@
...
 @router.get(
-    "/", tags=["get"], responses={
-        status.HTTP_200_OK: {
+    "/", tags=["get"],
+    responses={
+        200: {
             "model": ReturnShop
         },
-        status.HTTP_404_NOT_FOUND: {
+        404: {
```
Comment (Vghxv) [inline router/shop.py]: any particular reason why use pure number?

File: router/shop.py [hunk 5]
```diff
@@ -1,128 +1,106 @@
...
+    if result:
+        data.update({"data": result})
+        message = "Success"
+    else:
+        message = "Fail"
+
+    content = jsonable_encoder({"msg": message, **data})
+
+    return JSONResponse(status_code=status_code, content=content)
+
```
Comment (Vghxv) [inline router/shop.py]: function `create_shop` and `update_shop` , `if (result)` is evaluated inside and outside `prepare_response` which is kinda smelly.
More importantly, in function `get_shop`, `prepare_response` can return JSONResponse(status_code=status.HTTP_200_OK, content=content) when result is `false`

**Review by Vghxv — CHANGES_REQUESTED** [review]
> What does a 500 Internal Server Error mean?
> 
> A 500 Internal Server Error occurs when a server is unable to fulfill a request sent to it due to an unexpected blocker. The server then sends that error message to indicate there’s an issue with it but it can’t seem to diagnose the problem.
in our Backend `server`, i don't think  if statements is a `unexpected blocker`

File: router/shop.py (hunk 4 above)
Comment (PEKKA1117) [reply router/shop.py]: forgot to import lib or the JBAI tried to fuck me, i guess.

**Comment by Dragonfly911117** [comment]
Shit, i forgot to do auth-things to product. Will do it like on the bus to school.

**Review by Vghxv — APPROVED** [review]
lgtm

**Pull request merged**